### PR TITLE
Popup window

### PR DESCRIPTION
**Pull request by Amazinggracee**
- [ ] Use basic JavaScript syntax.
- [ ] Use JavaScript to manipulate DOM elements.
- [ ] Use JavaScript events.
- [ ] Understand how to parse a Figma design to create a UI.
- [ ] Use objects to store and access data.

**Project requirements**

- [ ] When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- [ ] When the user clicks (or taps) the close (X) button, the popup disappears.
- [ ] You need to use a JavaScript array to store all of the information for all projects.
- [ ] For each project you need to store the following pieces of data in a JavaScript object, at a minimum: name, description, 
        featured image, technologies, link to live version and link to the source.
- [ ] You need to update the main page so that the projects section is created dynamically using the information stored in that 
- [ ] JavaScript object. Remember, all of the HTML in that section is created when the page loads.
- [ ] After that, you could implement the popup window.
- [ ] You should implement popups for both mobile and desktop screen sizes

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  const h2 = document.createElement('h2');
+  if (window.innerWidth <= 768) {
+    div.classList.add('grid', 'mobile');
+    h2.textContent = projects[i].nameMobile;
+  } else {
+    div.classList.add('grid', 'desktop', projects[i].DesktopItem);
+    h2.textContent = projects[i].nameDesk;
+  }
+  div.appendChild(h2);
+}
```
Comment (prolajumokeoni) [inline script.js:112]: - [ ] Currently on the mobile screen, only the first is displayed, please add the other cards and their modal
- [ ] Please style your buttons to look like the Figma design

https://user-images.githubusercontent.com/69638013/210957908-d1211693-3cb9-4919-a56a-1acfe5151e93.mp4

File: script.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+// function called when see project button is clicked;
+function modal() {
+  document.body.appendChild(modalDisplay);
+  allElement.classList.add('bluring');
+  document.querySelector('#modal-close').addEventListener('click', () => {
+    modalDisplay.remove();
+    allElement.classList.remove('bluring');
+  });
+}
```
Comment (prolajumokeoni) [inline script.js:149]: - [ ] Kindly make sure that the data for each card is prepopulated in the modal, currently when I click on a modal for a card it does not show the same data of the card, it shows the same data for all cards

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Team,
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 - Kindly ensure that as partners you always have balanced commits in a pair programming project 👍
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  const h2 = document.createElement('h2');
+  if (window.innerWidth <= 768) {
+    div.classList.add('grid', 'mobile');
+    h2.textContent = projects[i].nameMobile;
+  } else {
+    div.classList.add('grid', 'desktop', projects[i].DesktopItem);
+    h2.textContent = projects[i].nameDesk;
+  }
+  div.appendChild(h2);
+}
```
Comment (roshan-bajgain) [inline script.js:118]: - [ ] As requested by the previous reviewer, In your mobile version only the first card is displaying please try to display all other cards. See the image below for reference
### Your design
his

### Figma design with six missing cards
Screenshot 2023-01-06 180751

File: script.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+// function called when see project button is clicked;
+function modal() {
+  document.body.appendChild(modalDisplay);
+  allElement.classList.add('bluring');
+  document.querySelector('#modal-close').addEventListener('click', () => {
+    modalDisplay.remove();
+    allElement.classList.remove('bluring');
+  });
+}
```
Comment (roshan-bajgain) [inline script.js:155]: - [ ] As requested by the previous reviewer, your card is showing the same image, buttons, and content for every card when it is popup but when the user clicks on the `see project` then the pop-up should appear as the same of that card. please implement it.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    DesktopItem: 'item13',
+    featuredImage: './Images/Desktop-modal.png',
+    mobileModalHeading: 'Multi Post Stories',
+    desktopModalHeading: 'Keeping track of hundreds  of components website',
+    modalDescription: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent",
+    live: 'https://github.com/Amazinggracee/Portfolio/',
+    source: 'https://github.com/Amazinggracee/Portfolio/window-pop',
+  },
+];
+
```
Comment (roshan-bajgain) [inline script.js:102]: - [ ] [Optional] You should stick to the design as much as possible please fix the buttons and spacing of content. See image below for reference
### Your design in the mobile version
your

### Figma design
fimga

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Amazinggracee, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @Amazinggracee, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

 - No linter errors ✔️ 
 - GitHub flow was correctly used ✔️ 
 - Good PR title and Description ✔️

## Aspects to improve ♻️

- [x] Check the comments under the review.
- [ ] As the previous code reviewer commented when switching from desktop to mobile or from mobile to desktop view, the cards disappear from the screen. Please fix this bug since this can make a bad experience for the user.
- [ ] As the previous code reviewer mentioned currently the data that is being displayed in the modal is not the data from the same object in the array of projects in your javascript file. Please fix this. The reason for using the javascript file to keep the data of your project is to make it easier to display and add new projects to your portfolio and the popup should represent the data of the project card the user clicked.
- 
Screen Recording 2023-01-06 at 6 03 39 PM


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

 - [OPTIONAL] In the video above is possible to see weird behavior while scrolling over the cards you created. This is a bad user experience. It would be nice to fix that to make a smooth hover effect. 
 - [OPTIONAL] There are also 2 cards that do not appear with the option to see the popup window. It would be great if all project has the possibility to open that.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — COMMENTED** [review]
Hello @Amazinggracee 👋,

_This is an additional review due to the second opinion request._

- Nice implementation of the modal popup functionality 👍 

  - I strongly suggest you implement the previously requested changes, currently, the mobile version of the web app is displaying only one project card while the Figma template design has a total of seven project cards.
  


https://user-images.githubusercontent.com/66112760/211106701-324be043-3914-4ed0-b9fe-16c865947921.mov



- When displaying the popup for each card, the popup data should be populated with the data from the projects array, i.e the popup for each project should display the exact data from the project card.


Keep rocking 🚀!

**Comment by Rudy-Menekam** [comment]
Thank you @bimbolabuari for the review, what @Amazinggracee did is exactly what is on the figma design, all six cards are the same. Unless the change you are requesting for is on a different figma design
This is the link to the figma design we are using https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471&t=I82gUKpVubL7sNTE-0

**Review by bimbolabuari — COMMENTED** [review]
Hello @Amazinggracee 👋,

_This is an additional review due to the second opinion request._

- The modal popup data should be populated with the data from the exact project card that is been clicked on, i.e the project title, description and image should match the pop-up data.



https://user-images.githubusercontent.com/66112760/211113458-dda4e70d-2494-4ff8-934f-ce6f85700bbe.mov



- According to the screen record above, when I click on the project card with the title "Website protfolio", the popup data title should be "Website protfolio" and not "Keeping track of hundreds of components website", this applies to the project description and image.

- When switching from different screen sizes(desktop to mobile or vice versa), the cards disappear from the screen, you have just one project card in the project section.


Screenshot 2023-01-07 at 12 05 23 AM


 If this looks otherwise on your side locally, you can make a video or screenshot based on this and this change will be overlooked by the next CR.
    
  - I hope have been able to make the previously requested changes clear enough 🙂 

Kindly make the requested changes and ask for a review on your dashboard 👍 

Keep rocking 🚀!

**Pull request merged**